### DISCUSSION TITLE
Why can't I make navigationLink normally without using IfLetStore???

### DISCUSSION THREAD

**Discussion by Yudi26** [pointfreeco/swift-composable-architecture#1048] [Q&A] [answered]
``` Swift
NavigationLink(LocalizedStringKey.init("Hy"), destination: Text("Hiiii"), isActive: viewStore.binding(get: $0.canLogin, send: { LoginAction.validateCredentials() }))
                    Button("Login") {
                        viewStore.send(.validateCredentials)
                    }.frame(width: 200, height: 40, alignment: .center)
                        .foregroundColor(.black)
                        .background(!viewStore.canLogin ? .green.opacity(0.3): .green.opacity(1))
                        .disabled(!viewStore.canLogin)
                        .cornerRadius(20)
                        .padding()
```

### When writing above code its not working in TCA. Please help me out...
**Getting Error:** 
```Swift
Contextual closure type '() -> TupleView<(NavigationLink<EmptyView, Text>, Button<Text>)>' expects 0 arguments, but 1 was used in closure body
```

**Comment by tgrapperon** [comment, marked as answer]
Hi,
It looks like a standard Swift error message. It seems that `$0.canLogin` misses its enclosing curly braces:`{ $0.canLogin }`. It also seems that you're using [the deprecated version](https://developer.apple.com/documentation/swiftui/navigationlink/init(_:destination:isactive:)-4whx5) of [this initializer ](https://developer.apple.com/documentation/swiftui/navigationlink/init(_:isactive:destination:)-gn8q) that you should preferably use.

Alternatively, you could probably use `\.canLogin` instead `{ $0.canLogin }`, because the `KeyPath<State, Bool>` is automatically promoted as a function `(State) -> Bool` equivalent to `{ $0.canLogin }`. That's maybe where things were mixed up.

Also, `LocalizedStringKey` is `ExpressibleByStringLiteral`, so you should directly use `"Hy"`.

By the way, you can preserve your code formatting by using a triple "`" instead of a single one, and add "swift" to get code highlighting:
````
```swift
let x = 2
```
````

I hope it'll help!

**Reply by Yudi26** [reply]
Thanks @tgrapperon for helping me out with very descriptive answer....

**Reply by Yudi26** [reply]
Hi @tgrapperon I need one more help here..
Now I can navigate and see "Hiiii" in Next Screen..

**Issue:**  When I navigate back to **First Screen** Then I can't navigate again to **Next Screen**("Hiiii").
**What could be cause of that?**

**Reply by tgrapperon** [reply]
Hello @Yudi26, sorry for not reaching out sooner.
You should check that your `canLogin` flag is reset when going back to the first screen. You can put a `.debug()` at the end of your reducer to activate the debug mode and see current actions and state changes in the console.
You should make sure that the `LoginAction.validateCredentials()` (which is a little odd with its void argument) is processed when going back, flipping back the `canLogin` flag to `false`.

It would really help if you could post your reducer's code, or even better, the whole feature: State+Action+Reducer+View.

**Reply by Yudi26** [reply]
Thanx a lot for such amazing🤩 answer.. I made `LoginAction.validateCredentials()` with argument like: `LoginAction.loginButtonTapped($0)` and updated status of `canLogin` and it worked like boom....
I have spent long hours on this issue...
